tuta-sdk/rust/sdk/src/crypto/crypto_facade.rs

```diff
@@ -32,22 +33,24 @@ const BUCKET_KEY_FIELD: &str = "bucketKey";
 
 #[derive(uniffi::Object)]
 pub struct CryptoFacade {
-	key_loader_facade: Arc<KeyLoaderFacade>,
+	key_loader_facade: Option<Arc<KeyLoaderFacade>>,
```

Comment:
I also don't really like it. I think we can construct a proof that makes it impossible to call the methods requiring the KeyloaderFacade unless you successfully logged in :thinking:

Reply:
It is one thing to do unauthenticated requests (that is kinda hidden in headers provider) but another to request encrypted things. We have multiple `EntityClient`s partially for this reason.

I really think you need two top-level `ServiceExecutor`s, maybe they both delegate to the same thing under the hood though.

Reply:
we had a long think about this and we came up with a few ideas to prevent mistakes at compile time, but nothing that we would implement now since it would require some more work.

We did a version that wraps a service executor that can resolve keys around a "normal" service executor though to not damage any invariants that were there before.